File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/cost/SapCostService.java
package top.bettercode.summer.tools.sap.cost;


import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import top.bettercode.summer.tools.sap.connection.SapService;

public class SapCostService {

  private final Logger log = LoggerFactory.getLogger(SapCostService.class);
  private final SapService sapService;

  public SapCostService(SapService sapService) {
    this.sapService = sapService;
  }

  public CostResp getCosts() {
    return sapService.invoke("ZRFC_OA_COST", null, CostResp.class);
  }

}

File: tools/sap/src/main/kotlin/top/bettercode/summer/tools/sap/cost/SapCostService.kt
package top.bettercode.summer.tools.sap.cost

import top.bettercode.summer.tools.sap.connection.SapService

class SapCostService(private val sapService: SapService) {
    val costs: CostResp
        get() = sapService.invoke("ZRFC_OA_COST", null, CostResp::class.java)
}
